### ISSUE TITLE
Does anyone have an error when building like me?

### ISSUE THREAD

**Issue by bitdessert** [mattermost/desktop#2378]
I confirm (by marking "x" in the [ ] below: [x]):

- [ ] This is not a troubleshooting question. [Troubleshooting questions go here: https://docs.mattermost.com/install/troubleshooting.html](https://docs.mattermost.com/install/troubleshooting.html).
- [ ] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [ ] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md).

---

**Summary**

The Windows version does not build.

**Environment**
- Operating System: macOS Ventura 13.0.1
- Mattermost Desktop App version: 5.2.0
- Mattermost Server version: 

**Steps to reproduce**
After downloading, try to build through the command below.
npm run package:windows

**Expected behavior**
The build should be complete.

**Observed behavior**
An error occurs during the rebuild process.
"rebuilding native dependencies  dependencies=macos-notification-state@2.0.1, windows-focus-assist@1.3.0 platform=win32 arch=ia32"

**Possible fixes**

**Comment by devinbinnie** [comment]
@bitdessert Which version of Node are you using? We recommend using the latest LTS version (v18.12.0 at time of writing)

**Comment by bitdessert** [comment]
@devinbinnie The node version is version 16 as shown in the contents.
The same issue occurred in version 18, so I downgraded.
I wondered if it was a Python problem, so I also upgraded Python.
The npm version also started testing at version 9, and downgraded.

**Comment by bitdessert** [comment]
This problem seems to be a compatibility issue with xcode.
It seems that xcode should use 12.5.1 or lower version, and macOS seems to use monterey or lower version.
Even if xcode is version 12.5.1 or lower, other problems occur when building in ventura version.
Also, even if it is a monterey version, it cannot be built with the same issue as the above from xcode 13 version.

**Comment by devinbinnie** [comment]
> @devinbinnie Can you tell me the environment where the build was successful? My environment is macOS Monterey (12.5.1), node (18.12.1), npm (8.19.2), xcode (12.5.1), clang (12.0.5), etc. In this case, the build succeeds, but a warning window pops up saying that notificationstate.node is not an appropriate win32 application in Windows OS and does not run.
> 
> As a result of trying to build by changing only xcode to 13.4.1 without modifying other environments, a rebuild error related to node-gyp occurred, and the issue is the same as the one raised above.
> 
> When executing the `node node-gyp rebuild` command, the error 'binding.gyp cannot be found' may appear.
> 
> Node was installed with `homebrew`.

That all looks very simillar to what I have. As for the `.node` issue, are you on the latest master? Also are you building for Windows on macOS? Because that won't work with native modules, they need to be built on the OS they are intended for.

**Comment by bitdessert** [comment]
@devinbinnie Right. Both Windows apps and macOS apps have been packaged in macOS. Are you saying that mac apps should be packaged on macOS, and windows apps should be built on windows OS? It didn't matter until version 5.1.0, but do I have to build it separately because of the macos-notification-state library and windows-focus-assist from 5.2?

**Comment by bitdessert** [comment]
@devinbinnie Unfortunately, I just tried building it in a windows environment and it failed, but I know how to do it, so I'll try building it and ask if there are any issues, thanks :)

**Comment by devinbinnie** [comment]
@bitdessert Is this still an issue for you?

**Comment by charanjeet-thind** [comment]
> @devinbinnie Unfortunately, I just tried building it in a windows environment and it failed, but I know how to do it, so I'll try building it and ask if there are any issues, thanks :)

I am facing same  issue on windows, can u pls guide me, how can i fix
F:\boomea\matermost2\desktop\node_modules\windows-focus-assist\lib\focus-assist.cc(156,5): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc [F:\boomea\matermost2\desktop\node_modules\windows-focus-assist\build\focusassist.vcxproj]
